controllers/greptimedbcluster/deployers/monitoring.go

```diff
@@ -0,0 +1,246 @@
+			klog.Errorf("Failed to update status: %s", err)
+		}
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func (d *MonitoringDeployer) createPipeline(cluster *v1alpha1.GreptimeDBCluster) error {
+	var b bytes.Buffer
+	w := multipart.NewWriter(&b)
+
+	fw, err := w.CreateFormFile("file", "pipeline.yaml")
+	if err != nil {
+		return err
+	}
+
+	pipeline, err := d.defaultPipeline()
+	if err != nil {
+		return err
+	}
+
+	// If the pipeline is specified in the CR, use that instead.
+	if p := cluster.GetMonitoring().GetLogsCollection().GetPipeline().GetData(); p != "" {
+		pipeline = p
+	}
+
+	_, err = io.Copy(fw, strings.NewReader(pipeline))
+	if err != nil {
+		return err
+	}
+	w.Close()
+
+	standaloneName := common.ResourceName(common.MonitoringServiceName(cluster.Name), v1alpha1.StandaloneKind)
+
+	// FIXME(zyy17): Make the port configurable.
+	svc := fmt.Sprintf("%s.%s.svc.cluster.local:%d", standaloneName, cluster.Namespace, v1alpha1.DefaultHTTPPort)
+	hc := &http.Client{
+		Timeout: 5 * time.Second,
+	}
+
+	operation := func() error {
+		req, err := http.NewRequest("POST", fmt.Sprintf("http://%s/v1/events/pipelines/%s", svc, common.LogsPipelineName(cluster.Namespace, cluster.Name)), &b)
+		if err != nil {
+			return err
+		}
+		req.Header.Set("Content-Type", w.FormDataContentType())
+
+		resp, err := hc.Do(req)
+		if err != nil {
+			klog.Warningf("failed to create pipeline: %v", err)
+			return err
+		}
+		defer resp.Body.Close()
+
+		responseBody, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+
+		if resp.StatusCode != http.StatusOK {
+			return fmt.Errorf("failed to create pipeline: '%s'", string(responseBody))
+		}
+
+		return nil
+	}
+
+	// The server may not be ready to accept the request, so we retry a few times.
+	if err = retry.Do(
+		operation,
+		retry.Attempts(10),
+		retry.Delay(500*time.Millisecond),
+		retry.DelayType(retry.FixedDelay),
+	); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// defaultPipeline returns the default pipeline that will be used by the standalone greptimedb instance to collect greptimedb logs.
+func (d *MonitoringDeployer) defaultPipeline() (string, error) {
+	data, err := fs.ReadFile(config.DefaultPipeline, "pipeline.yaml")
+	if err != nil {
+		return "", err
+	}
+	return string(data), nil
+}
+
+var _ deployer.Builder = &monitoringBuilder{}
+
+type monitoringBuilder struct {
+	*CommonBuilder
+}
+
+func (b *monitoringBuilder) BuildGreptimeDBStandalone() deployer.Builder {
+	if b.Cluster.GetMonitoring() == nil || b.Cluster.GetMonitoring().GetStandalone() == nil {
+		return b
+	}
+
+	if b.Err != nil {
+		return b
+	}
+
+	standalone := &v1alpha1.GreptimeDBStandalone{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "GreptimeDBStandalone",
+			APIVersion: "greptime.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      common.MonitoringServiceName(b.Cluster.Name),
+			Namespace: b.Cluster.Namespace,
+		},
+		Spec: *b.Cluster.GetMonitoring().GetStandalone().DeepCopy(),
```

Comment:
Why the greptimedbcluster directory need to add the greptimedbstandalone configuration?

Reply:
Because the standalone will use the same storage with cluster.